daemon/cmd/kube_proxy_replacement.go

```diff
@@ -67,12 +67,7 @@ func initKubeProxyReplacementOptions() (bool, error) {
 		option.Config.EnableSocketLB = false
 		option.Config.EnableHostServicesTCP = false
 		option.Config.EnableHostServicesUDP = false
-
-		if option.Config.EnableSessionAffinity {
-			if err := disableSessionAffinityIfNeeded(true); err != nil {
-				return false, err
-			}
-		}
```

Comment:
Users can still enable session affinity for E-W traffic when kpr is disabled. See this note in the kpr guide - 

`For users who run with kube-proxy (i.e. with Cilium’s kube-proxy replacement disabled), the ClusterIP service loadbalancing when a request is sent from a pod running in a non-host network namespace is still performed at the pod network interface (until [GH#16197](https://github.com/cilium/cilium/issues/16197) is fixed). For this case the session affinity support is disabled by default. To enable the feature, set config.sessionAffinity=true.`

Reply:
thanks, thought I was doing something fishy. reverted (and guarded the probe in disableSessionAffinityIfNeeded).